components/ValidInsurancePicker.js

```diff
@@ -27,17 +30,35 @@ const ValidInsurancePicker = () => {
     setIsModalVisible(false);
   };
 
+  const renderValidInsuranceButton = (buttonText = "") => {
+    return (
+        <TouchableOpacity
+          style={styles.button}
+          onPress={openModal}
+        >
+          <Text style={styles.buttonText}>{buttonText}</Text>
+        </TouchableOpacity>
+    )
+}
+
+// Refactored 
+const renderValidInsurance = (hasValidInsurance = false) => {
+  return hasValidInsurance
+    ? renderValidInsuranceButton(insuredStatus)
+    : renderValidInsuranceButton(setInsuredStatus(APP_STRINGS.no));
```

Comment:
I think that this `setInsuredStatus(APP_STRINGS.no);` can go right before the ternary inside the function.

Was this the thing we talked about on chat today, or was it something else? It's one of these ternary functions that calls a function that returns JSX, and we had to set component state first, you remember?

Reply:
I believe so... The example we worked through was breaking if the user pushed "Save and Continue" without setting the input... I'm gonna go over it manana and investigate.